application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -224,4 +248,74 @@ private boolean isHelpThread(MessageReceivedEvent event) {
         return isStagingChannelName.test(rootChannelName)
                 || isOverviewChannelName.test(rootChannelName);
     }
+
+    @Override
+    public String getName() {
+        return "filesharing";
+    }
+
+    @Override
+    public void acceptComponentIdGenerator(ComponentIdGenerator generator) {
+        componentIdInteractor.acceptComponentIdGenerator(generator);
+    }
+
+    @Override
+    public UserInteractionType getInteractionType() {
+        return UserInteractionType.OTHER;
+    }
+
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
+        Member user = event.getMember();
+        String authorId = args.get(0);
+        boolean hasSoftModPermissions =
+                user.getRoles().stream().map(Role::getName).noneMatch(softModPattern);
+
+        if (!authorId.equals(user.getId()) && hasSoftModPermissions) {
+            event.reply("You do not have permission for this action.").setEphemeral(true).queue();
+            return;
+        }
+
+        String gistId = args.get(1);
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(SHARE_API + "/" + gistId))
+            .header("Accept", "application/json")
+            .header("Authorization", "token " + gistApiKey)
+            .DELETE()
+            .build();
+
+        HttpResponse<String> apiResponse;
+        try {
+            apiResponse = CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException(
+                    "Attempting to delete a gist, but the request got interrupted.", e);
+        }
+
+
+        int status = apiResponse.statusCode();
+        if (status == 404) {
+            throw new IllegalStateException("Gist API unexpected response while deleting gist: %s."
+                .formatted(apiResponse.body()));
+        }
+
+        Message message = event.getMessage();
+        List<Button> buttons = message.getButtons();
+        event.editComponents(ActionRow.of(buttons.stream().map(Button::asDisabled).toList()))
+            .queue();
```

Comment:
@Tais993 i might be wrong, but do we need to `deferReply()` in an onButtonClick event? if so, this should be deferred. its doing a blocking request to github api and waits for the result.

Reply:
gonna wait for response before changing something

Reply:
just checked, yes its needed. so as soon as ur past the basic checks and right before u start the long operations (the api call), u have to do `event.deferReply().queue();` and then, from there on, u must only do interactions through the hook. that is, u cant use `event.editComponents` anymore, but `event.getHook().editComponents(...)` and similar.

Reply:
(actually i have a better idea - see new conversation down below)